src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -269,6 +309,12 @@ public abstract class GitSourceProvider : ISourceProvider
             }
 
             bool exposeCred = StringUtil.ConvertToBoolean(executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.PersistCredentials));
+            
+            // Read 'fetch by commit' value from the execution variable first, then from the environment variable if the first one is not set
+            string fetchByCommitExecution = executionContext.Variables.GetValueOrDefault("agent.source.git.fetchbycommit")?.Value;
+            bool fetchByCommit = fetchByCommitExecution != null ?
```

Comment:
what's the default for `fetchByCommit` if i didn't provide variable in executionConext and environment variable?
i think `fetchByCommit` should get enabled by default, with your current logic, i think the `fetchByCommit` will get disabled by default.

change the variable to `VSTS.DisableFetchByCommit` and `VSTS_DISABLEFETCHBYCOMMIT`

```C#
bool fetchByCommit = StringUtil.ConvertToBoolean(executionContext.Variables.GetValueOrDefault("vsts.disableFetchByCommit")??Environment.GetEnvironmentVariable("VSTS_DISABLEFETCHBYCOMMIT"), true);
```

Reply:
Originally, I had some concerns about possible functional regressions, back when this PR also implemented "fetch by ref", so I started with default off. However, now that it's only scoped to "fetch by commit", I'm good with turning it on by default.